### DISCUSSION TITLE
addTimer 为什么是线程安全的，通过什么写法实现的？

### DISCUSSION THREAD

**Discussion by sychen12345** [chenshuo/muduo#662] [General]
陈硕老师，请问下，你的定时器添加 addTimer()和addTimerInLoop()， 是怎么确保线程安全的，除了加了assertInLoopThread()，是io线程的检测，我不太理解，怎么确保这个io线程才能调用这两个方法，我如果一定要用   其他线程loop->addTimerInLoop()呢？

**Comment by chenshuo** [comment]
你可以先试试能不能复现你提到的问题。
如果能复现，那么我们可以针对具体的复现代码来讨论如何修复。
如果不能复现，那问题就可以认为不存在了。
